snapcraft/internal/build_providers/_snap.py

```diff
@@ -0,0 +1,296 @@
+                "snap-revision",
+                "snap-revision={}".format(snap_info["revision"]),
+                "snap-id={}".format(snap_info["id"]),
+            ]
+        )
+        return assertions
+
+    def _set_data(self) -> None:
+        op = self.get_op()
+        snap_repo = self._get_snap_repo()
+        install_cmd = ["sudo", "snap"]
+
+        if op == _SnapOp.INJECT:
+            install_cmd.append("install")
+            snap_info = snap_repo.get_local_snap_info()
+            snap_revision = snap_info["revision"]
+
+            if snap_revision.startswith("x"):
+                install_cmd.append("--dangerous")
+
+            if snap_info["confinement"] == "classic":
+                install_cmd.append("--classic")
+
+            snap_file_name = "{}_{}.snap".format(snap_repo.name, snap_revision)
+            install_cmd.append(os.path.join(self._snap_dir, snap_file_name))
+        elif op == _SnapOp.INSTALL or op == _SnapOp.REFRESH:
+            install_cmd.append(op.name.lower())
+            snap_info = snap_repo.get_store_snap_info()
+            snap_revision = snap_info["channels"]["latest/stable"]["revision"]
+            confinement = snap_info["channels"]["latest/stable"]["confinement"]
+            if confinement == "classic":
+                install_cmd.append("--classic")
+            install_cmd.append(snap_repo.name)
+        elif op == _SnapOp.NOP:
+            install_cmd = []
+            snap_revision = None
+
+        self.__install_cmd = install_cmd
+        self.__revision = snap_revision
+
+    def get_revision(self) -> str:
+        if self.__revision is None:
+            self._set_data()
+        return self.__revision
+
+    def get_install_cmd(self) -> List[str]:
+        if self.__install_cmd is None:
+            self._set_data()
+        return self.__install_cmd
+
+
+_STORE_ASSERTION = [
+    "account-key",
+    "public-key-sha3-384=BWDEoaqyr25nF5SNCvEv2v7QnM9QsfCc0PBMYD_i2NGSQ32EF2d4D0hqUel3m8ul",
+]
+
+
+class SnapInjector:
+    def __init__(
+        self,
+        *,
+        snap_dir: str,
+        registry_filepath: str,
+        runner,
+        snap_dir_mounter,
+        snap_dir_unmounter,
+        file_pusher
+    ) -> None:
+
+        self._snaps = []  # type: List["_SnapManager"]
+        self._snap_dir = snap_dir
+        self._registry_filepath = registry_filepath
+        self._runner = runner
+        self._snap_dir_mounter = snap_dir_mounter
+        self._snap_dir_unmounter = snap_dir_unmounter
+        self._file_pusher = file_pusher
+
+        self._registry_data = dict()  # type: Union[None, Dict[str, List[Any]]]
+
+    def _load_registry(self):
+        if self._registry_filepath is None or self._registry_data:
+            return
+        if not os.path.exists(self._registry_filepath):
+            return
+
+        with open(self._registry_filepath) as registry_file:
+            self._registry_data = yaml.load(registry_file)
+
+    def _dump_registry(self):
```

Comment:
Take it or leave it, but I feel like this should be called `_save_registry`.

Reply:
ok, was going with parity with yaml and json.load/dump :-)